File: finance-app/src/app/core/models/transaction.dto.ts
import type { ClassificationStatusDto } from './classification-status.dto';
import type { MerchantDto } from './merchant.dto';
import type { MoneyDto } from './money.dto';
import type { TransactionSourceDto } from './transaction-source.dto';

export interface TransactionDto {
  id: string;
  amount: MoneyDto;
  occurredAt: Date;
  description: string | null;
  merchantId: string | null;
  merchant: MerchantDto | null;
  source: TransactionSourceDto;
  classificationStatus: ClassificationStatusDto;
}

File: finance-core/FinanceCore.Application/Transactions/Dtos/TransactionDto.cs
using FinanceCore.Domain.Transactions;

namespace FinanceCore.Application.Transactions.Dtos;

public sealed record TransactionDto(
    Guid Id,
    MoneyDto Amount,
    DateTimeOffset OccurredAt,
    string? Description,
    Guid? MerchantId,
    MerchantDto? Merchant,
    TransactionSource Source,
    ClassificationStatusDto ClassificationStatus);
